services/player-service/application/src/test/java/com/kjeldsen/player/application/usecases/GenerateDeclinePhaseUseCaseTest.java

```diff
@@ -0,0 +1,99 @@
+package com.kjeldsen.player.application.usecases;
+
+import com.kjeldsen.events.domain.EventId;
+import com.kjeldsen.player.domain.Player;
+import com.kjeldsen.player.domain.events.PlayerTrainingDeclineEvent;
+import com.kjeldsen.player.domain.provider.InstantProvider;
+import com.kjeldsen.player.domain.repositories.PlayerReadRepository;
+import com.kjeldsen.player.domain.repositories.PlayerTrainingDeclineEventWriteRepository;
+import com.kjeldsen.player.domain.repositories.PlayerWriteRepository;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import java.time.Instant;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+class GenerateDeclinePhaseUseCaseTest {
+
+    private static final Integer CURRENT_DAY = 7;
+    private static final Integer DECLINE_SPEED = 56;
+    private static final Player.PlayerId PLAYER_ID = mock(Player.PlayerId.class);
+    private static final EventId EVENT_ID = EventId.from("event-id");
+    private static final Instant NOW = Instant.now();
+
+    private final PlayerReadRepository mockedPlayerReadRepository = Mockito.mock(PlayerReadRepository.class);
+    private final PlayerWriteRepository mockedPlayerWriteRepository = Mockito.mock(PlayerWriteRepository.class);
+    private final PlayerTrainingDeclineEventWriteRepository mockedPlayerTrainingDeclineEventWriteRepository = Mockito.mock(PlayerTrainingDeclineEventWriteRepository.class);
+
+    private final GenerateSingleDeclineTrainingUseCase generateDeclinePhaseUseCase = new GenerateSingleDeclineTrainingUseCase(mockedPlayerTrainingDeclineEventWriteRepository, mockedPlayerReadRepository, mockedPlayerWriteRepository);
+
+    @Test
+    @DisplayName("Generate should throw if player not found.")
+    void generate_should_throw_exception_if_player_not_found() {
+        // Arrange
+        Player.PlayerId playerId = new Player.PlayerId("test");
+
+        when(mockedPlayerReadRepository.findOneById(playerId)).thenReturn(Optional.empty());
+
+        // Act & Assert
+        assertThrows(RuntimeException.class, () -> generateDeclinePhaseUseCase.generate(PLAYER_ID, CURRENT_DAY, DECLINE_SPEED));
```

Comment:
check message

Reply:
I made the same thing as in the previous comment.